File: ocs_samples/library_samples/Java/ocs_sample_library_preview/src/main/java/com/github/osisoft/ocs_sample_library_preview/dataviews/DataviewQueryQuery.java
/** DataviewQueryQuery.java
 * 
 */

package  com.github.osisoft.ocs_sample_library_preview.dataviews;


public class DataviewQueryQuery {

    private String Resource = "";
    private String Field = "";
    private String Value = "";
    private String Function = "";

    /**
     * base constructor
     */
    public DataviewQueryQuery()
    {
    }

    /**
     * generate the query from passed in values
     * 
     * @param Resource query resource can be something like "Streams", "TypeProperties"   required
     * @param Field query field can be something like "Id", "Name", "Tag", "Description", "TypeId", "MetadataKey"   required
     * @param Value value for field to use in query   required
     * @param Function  QueryFunction can be something like "Contains", "Equals", "EndsWith", "StartsWith"    required
     */
    public DataviewQueryQuery(String resource, String field, String Value, String Function)
    {
        this.Resource =resource;
        this.Field =field;
        this.Value = Value;
        this.Function = Function;
    }

    /**
     * gets resource   query resource can be something like "Streams", "TypeProperties" 
     * @return
     */
    public String getResource() {
        return Resource;
    }

    /**
     * sets resource    query resource can be something like "Streams", "TypeProperties" 
     * @param resource
     */
    public void setResource(String resource) {
        this.Resource = resource;
    }

    /**
     * gets field   query field can be something like "Id", "Name", "Tag", "Description", "TypeId", "MetadataKey"  
     * @return
     */
    public String getField() {
        return Field;
    }

    /**
     * sets field   query field can be something like "Id", "Name", "Tag", "Description", "TypeId", "MetadataKey"  
     * @param field
     */
    public void setField(String field) {
        this.Field = field;
    }

    /**
     * get value   value for field to use in query 
     * @return
     */
    public String getValue() {
        return Value;
    }

    /**
     * sets value   value for field to use in query 
     * @param value
     */
    public void setValue(String value) {
        this.Value = value;
    }

    /**
     * gets function    QueryFunction can be something like "Contains", "Equals", "EndsWith", "StartsWith"  
     * @return
     */
    public String getFunction() {
        return Function;
    }

    /**
     * sets function    QueryFunction can be something like "Contains", "Equals", "EndsWith", "StartsWith"  
     * @param function
     */
    public void setFunction(String function) {
        this.Function = function;
    }
}

File: ocs_samples/library_samples/Python3/ocs_sample_library_preview/Dataview/DataviewQueryQuery.py
# DataviewQueryQuery.py
#

import json


class DataviewQueryQuery(object):

    def __init__(self, resource=None, field=None, value=None, function=None):
        self.__resource = resource
        self.__field = field
        self.__value = value
        self.__function = function

    """Sds dataview definition"""
    @property
    def Resource(self):
        """
        query resource can be something like "Streams", "TypeProperties"
                       required
        :return:
        """
        return self.__resource

    @Resource.setter
    def Resource(self, resource):
        """
        query resource   required
        :param resource:
        :return:
        """
        self.__resource = resource

    @property
    def Field(self):
        """
        query field can be something like "Id", "Name", "Tag",
               "Description", "TypeId", "MetadataKey"   required

        :return:
        """
        return self.__field

    @Field.setter
    def Field(self, field):
        """
        query field can be something like "Id", "Name"
               "Tag", "Description", "TypeId", "MetadataKey"   required
        :param field:
        :return:
        """
        self.__field = field

    @property
    def Value(self):
        """
        value for field to use in query   required
        :return:
        """
        return self.__value

    @Value.setter
    def Value(self, value):
        """
        value for field to use in query   required
        :param value:
        :return:
        """
        self.__value = value

    @property
    def Function(self):
        """
        QueryFunction can be something like "Contains", "Equals",
                     "EndsWith", "StartsWith"    required
        :return:
        """

        return self.__function

    @Function.setter
    def Function(self, function):
        """
        QueryFunction can be something like "Contains", "Equals",
                    "EndsWith", "StartsWith"    required
        :param function:
        :return:
        """
        self.__function = function

    def toJson(self):
        return json.dumps(self.toDictionary())

    def toDictionary(self):
        # required properties
        dictionary = {'Resource': self.Resource, 'Field': self.Field,
                      'Value': self.Value, 'Function': self.Function}

        return dictionary

    @staticmethod
    def fromJson(jsonObj):
        return DataviewQueryQuery.fromDictionary(jsonObj)

    @staticmethod
    def fromDictionary(content):
        dataviewQueryQuery = DataviewQueryQuery()

        if len(content) == 0:
            return dataviewQueryQuery

        if 'Resource' in content:
            dataviewQueryQuery.Resource = content['Resource']

        if 'Field' in content:
            dataviewQueryQuery.Field = content['Field']

        if 'Value' in content:
            dataviewQueryQuery.Value = content['Value']

        if 'Function' in content:
            dataviewQueryQuery.Function = content['Function']

        return dataviewQueryQuery
